janus_core/calculations/eos.py

```diff
@@ -246,27 +246,46 @@ def __init__(
         if not self.struct.calc:
             raise ValueError("Please attach a calculator to `struct`.")
 
-        if self.minimize and self.logger:
-            set_minimize_logging(
-                self.logger, self.minimize_kwargs, self.log_kwargs, track_carbon
-            )
+        set_minimize_logging(
+            self.logger, self.minimize_kwargs, self.log_kwargs, track_carbon
+        )
 
         # Set output files
-        self.write_kwargs.setdefault("filename", None)
         self.write_kwargs["filename"] = self._build_filename(
-            "generated.extxyz", filename=self.write_kwargs["filename"]
-        ).absolute()
+            "generated.extxyz", filename=self.write_kwargs.get("filename")
+        )
 
-        self.plot_kwargs.setdefault("filename", None)
         self.plot_kwargs["filename"] = self._build_filename(
-            "eos-plot.svg", filename=self.plot_kwargs["filename"]
-        ).absolute()
+            "eos-plot.svg", filename=self.plot_kwargs.get("filename")
+        )
+        self.fit_file = self._build_filename("eos-fit.dat")
+        self.raw_file = self._build_filename("eos-raw.dat")
 
         self.results = {}
         self.volumes = []
         self.energies = []
         self.lattice_scalars = empty(0)
 
+    @property
+    def output_files(self) -> None:
+        """
+        Dictionary of output file labels and paths.
+
+        Returns
+        -------
+        dict[str, PathLike]
+            Output file labels and paths.
+        """
+        return {
+            "log": self.log_kwargs["filename"] if self.logger else None,
```

Comment:
I certainly don't think this should be blocking, but I am half-wondering whether we want `@property`s for these which check if their respective trigger is valid to remove this pattern everywhere.

Not this PR, we should get this in.